src/node-jspm-jasmine.js

```diff
@@ -141,17 +180,73 @@ export function runTests(opts, errCallback = function() {}) {
 							importTheseTestFiles();
 						}
 					})
-					.catch(errCallback);
+					.catch((ex) => {
+						if (opts.coverage && opts.coverage.tempDirectory) {
+							// remove temporary directory
+							rimraf.sync(opts.coverage.tempDirectory)
+						}
+						errCallback(ex)
+					});
 				});
 			});
 		}
 	} catch(ex) {
-		removeTempFiles();
+		if (opts.coverage && opts.coverage.tempDirectory) {
+			// remove temporary directory
+			rimraf.sync(opts.coverage.tempDirectory)
+		}
 		errCallback(new Error(`Jasmine or Jspm may not be properly configured -- '${ex.toString()}'`));
 	}
 }
 
 function importTestFiles(SystemJS, jasmine, specDir, specFiles, coverage, errCallback) {
+
+	if (coverage) {
+		// the "onComplete" hook to prevent jasmine's self-righteous exit
+		jasmine.onComplete(function (passed) {
+			// avoid misalignment with jasmine's output
+			console.log('');
+			// import the rest of the modules not already imported (evaluated)
+			// as dependencies of specs
+			Promise.all(Object.keys(coverage.coverageFiles).map(function (file) {
+				return SystemJS.import(path.join(process.cwd(), file));
+			})).then(function () {
+				if (typeof __coverage__ === 'undefined') {
+					console.log("No coverage was collected for files matching globs " + coverage.files);
+				} else {
+					try {
+						const collector = remap(__coverage__)
+						let report = Report.create(
+							(coverage.reporter || 'html'),
+							{
+								dir: coverage.dir
+							}
+						);
+						report.writeReport(collector, true);
```

Comment:
@joeldenning Not a problem, but I don't think we are quite done here. Looking at this further, there is a concern that needs to be addressed in relation to the jspm issue. There are three things that could happen when we instrument files:
-  The source in the `SystemJS.instantiate` is NOT the same as the actual file source and sourcemap is provided. This has been addressed in the beginning.
-  The source in the `SystemJS.instantiate` IS the same as the actual file source and no sourcemap is provided. This was addressed some time ago.
-  The source in the `SystemJS.instantiate` is NOT the same as the actual file source and no sourcemap is provided. This is the issue at hand, causing this `TypeError: Cannot read property 'text' of undefined.` error.

Here's the tricky part with the issue at hand, Istanbul has to instrument the transpiled source, but the transpiled source only exists in memory (or in tmp), so I can't feed it the original source path because it's not going to find the transpiled source there. The net result is that you'll get a tmp folder in your coverage report, with no way to trace it back to the source. Ideally, instead of adapting to this issue, we should throw an error advising the user to fix his/her sourcemap situation with jspm, alternatively (included in this last pr update), I've switched the temp folder name form "tmp" to "no-source-map", which coveys the message that a sourcemap is missing, but the transpiled code was still processed. Another difference (which could be fixed with a bit of effort) is that instead of folders inside the html report, the user would see filenames with its path delimited by "|". What are your thoughts on this?